addons/web/static/src/core/utils/patch.js

```diff
@@ -118,3 +118,28 @@ export function unpatch(obj, patchName) {
         }
     }
 }
+
+export function applyOnRegistryItem(registry, itemKey, transformFn) {
+    if (!patchMap.has(registry)) {
+        patchMap.set(registry, {
+            transforms: [],
+            new: true,
+        });
+    }
+    const registryTransforms = patchMap.get(registry);
+
+    if (registryTransforms.new) {
+        registry.on("UPDATE", registry, (event) => {
+            if (event.operation === "add") {
+                for (const transform of registryTransforms.transforms) {
+                    if (registry.contains(transform.itemKey)) {
+                        transform.transformFn(registry.get(transform.itemKey));
+                    }
+                }
+            }
+        });
+    }
```

Comment:
when multiple patches will be applied on a same registry, this callback will be registered multiple times

Reply:
or not :-)
I misread your code :-D